mteb/evaluation/evaluators/ClassificationEvaluator.py

```diff
@@ -14,6 +19,66 @@
 logger = logging.getLogger(__name__)
 
 
+def dot_distance(a: np.ndarray, b: np.ndarray) -> float:
+    return -np.dot(a, b)
+
+
+class kNNMultiLabelClassificationEvaluator(Evaluator):
+    def __init__(
+        self,
+        embeddings_train,
+        y_train,
+        embeddings_test,
```

Comment:
Looking at the structure of `kNNClassificationEvaluator`, this new evaluator seems to be very different: encode isn't called here, test cache is omitted. Any particular reason for those?

Reply:
Yes consult the discussion on the PR, this is all done for performance reasons

Reply:
Btw, @KennethEnevoldsen is already assigned as reviewer and is well-versed in the decisions we made here I'm not sure two reviewers are necessary :)

Reply:
I see the comment now, thanks!